scripts/deploy-federation.sh

```diff
@@ -215,3 +219,17 @@ CONTEXT="$(kubectl config current-context)"
 for c in ${JOIN_CLUSTERS}; do
   ./bin/kubefed2 join "${c}" --host-cluster-context "${CONTEXT}" --add-to-registry --v=2 ${KF_NS_ARGS}
 done
+
+if [ "`uname`" == 'Darwin' ];then
+
+    # We need to fix cluster ip addr in cluster-registry for mac os.
+    # Assume all context was contained in current kubeconfig.
+    for c in `kubectl config get-contexts -o name`;
+    do
+        k=`kubectl config get-contexts ${c} --no-headers |awk -F "${c}" '{print $2}'|awk '{print $1}'`
+        ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${k}-control-plane)
+
+        kubectl patch clusters -n kube-multicluster-public ${c} --type merge \
+            --patch "{\"spec\":{\"kubernetesApiEndpoints\":{\"serverEndpoints\":[{\"clientCIDR\":\"0.0.0.0/0\", \"serverAddress\":\"https://${ip_addr}:6443\"}]}}}"
```

Comment:
Not sure, why this is required in mac os. there is no deviation from method used in Linux.

Reply:
https://github.com/kubernetes-sigs/federation-v2/blob/b7824b2b77472dae1f2097ebf7c9e656d497dcd0/scripts/create-clusters.sh#L123


See the comment.